src/typechecker.erl

```diff
@@ -747,9 +751,11 @@ int_type_to_range({type, _, integer, []})              -> {neg_inf, pos_inf};
 int_type_to_range({type, _, neg_integer, []})          -> {neg_inf, -1};
 int_type_to_range({type, _, non_neg_integer, []})      -> {0, pos_inf};
 int_type_to_range({type, _, pos_integer, []})          -> {1, pos_inf};
-int_type_to_range({type, _, range, [{integer, _, I1},
-                                    {integer, _, I2}]})
-                                        when I1 =< I2  -> {I1, I2};
+int_type_to_range({type, _, range, [{Tag1, _, I1}, {Tag2, _, I2}]})
+  when Tag1 =:= integer orelse Tag1 =:= char,
```

Comment:
it would be a bit weird from the user to mix integer and char within a range but mixed tags are allowed nevertheless